cpp/src/arrow/util/dispatch_internal.h

```diff
@@ -76,7 +76,8 @@ class DynamicDispatch {
 
  protected:
   // Use the Implementation with the highest DispatchLevel
-  void Resolve(const std::vector<Implementation>& implementations) {
+  template <typename Range>
+  void Resolve(const Range& implementations) {
```

Comment:
Can probably use a universal ref?
```suggestion
  void Resolve(Range&& implementations) {
```

Reply:
I don't think it changes anything since we are never moving the object inside the implementation.